tests/test_validate_outputs.py

```diff
@@ -0,0 +1,492 @@
+# This file is part of ci_hsc_gen3.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import os
+import unittest
+
+from lsst.daf.butler import Butler
+import lsst.utils.tests
+
+from lsst.utils import getPackageDir
+
+from lsst.ci.hsc.gen3 import DATA_IDS
+
+
+class TestValidateOutputs(lsst.utils.tests.TestCase):
+    """Check that ci_hsc_gen3 outputs are as expected."""
+
+    def setUp(self):
+        self.butler = Butler(os.path.join(getPackageDir("ci_hsc_gen3"), "DATA"),
+                             instrument="HSC", skymap="discrete/ci_hsc",
+                             writeable=False, collections=["HSC/runs/ci_hsc"])
+
+        self._num_exposures = len(DATA_IDS)
+        self._num_exposures_good_templates = 29
+        self._num_visits = len({data_id["visit"] for data_id in DATA_IDS})
+        self._num_tracts = 1
+        self._num_patches = 1
+        self._num_bands = len({data_id["physical_filter"] for data_id in DATA_IDS})
+        self._min_sources = 100
+
+    def assertTrueDescr(self, description, success):
+        if not success:
+            raise AssertionError(f"Failed test: {description}")
+
+    def assertEqualDescr(self, description, obj1, obj2):
```

Comment:
In fact I'd be a lot happier if we got rid of all of these and just use the standard `assertEqual(a, b, msg=x)` and `assertGreaterEqual()`.